contrib/ci/rustfmt.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2023 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+
+import glob
+import os
+import sys
+import subprocess
+
+
+def __get_license(fn: str) -> str:
+    with open(fn, "r") as f:
+        for line in f.read().split("\n"):
+            if line.find("SPDX-License-Identifier:") > 0:
+                return line.split(":")[1]
+    return ""
+
+
+def test_files() -> int:
+
+    rc: int = 0
+    for fn in glob.glob("**/*.rs", recursive=True):
+        if "meson-private" in fn:
+            continue
+        if "venv" in fn:
+            continue
+        if fn.startswith("subprojects"):
+            continue
+        if fn.startswith("dist"):
+            continue
+        print(fn)
+        try:
+            subprocess.check_output(["rustfmt", "--quiet", "--check", fn])
```

Comment:
So what's going to be interesting here is that not everyone will have `rustfmt` installed by default.  So two things this means:
1. Pre-commit CI job needs to explicitly install `rustc`/`rustfmt` if it's not doing this.
2. This test/script should be skipped if `rustfmt` isn't available.

Reply:
I did option 2.